docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,791 @@
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Register<TMessage> ( object, object, bool, Action<TMessage> )
+
+There is no direct replacement for this registration mechanism which allows you to support receiving messages for derived message types also. This change is intentional as the `Messenger` implementation aims to not use reflection to achieve it's performance benefits.
+
+Alternatively, there are a few options that can be done to achieve this functionality. 
+
+- Create a custom `IMessenger` implementation
+- Register the additional message types using a shared `Action`
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, nameof(MyViewModel), true, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+Messenger.Default.Register<MyOtherMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Send<TMessage> ( TMessage )
+
+The functionality of `Send<TMessage>(TMessage)` can be achieved with the MVVM Toolkit's `IMessenger` extension method `Send<TMessage>(TMessage)`. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Send<MyMessage>(new MyMessage());
+Messenger.Default.Send(new MyMessage());
+
+// Toolkit.Mvvm
+Messenger.Default.Send<MyMessage>(new MyMessage());
+Messenger.Default.Send(new MyMessage());
+```
+
+#### Send<TMessage> ( TMessage, object )
+
+The functionality of `Send<TMessage>(TMessage, object)` can be achieved with the MVVM Toolkit's `Send<TMessage, TToken>(TMessage, TToken)` method. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Send<MyMessage>(new MyMessage(), nameof(MyViewModel));
+Messenger.Default.Send(new MyMessage(), nameof(MyViewModel));
+
+// Toolkit.Mvvm
+Messenger.Default.Send(new MyMessage(), nameof(MyViewModel));
+```
+
+#### Unregister ( object )
+
+The functionality of `Unregister(object)` can be achieved with the MVVM Toolkit's `UnregisterAll(object)` method. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Unregister(this);
+
+// Toolkit.Mvvm
+Messenger.Default.UnregisterAll(this);
+```
+
+#### Unregister<TMessage> ( object )
+
+The functionality of `Unregister<TMessage>(object)` can be achieved with the MVVM Toolkit's `IMessenger` extension method `Unregister<TMessage>(object)`. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Unregister<MyMessage>(this);
+
+// Toolkit.Mvvm
+Messenger.Default.Unregister<MyMessage>(this);
+```
+
+#### Unregister<TMessage> ( object, Action<TMessage> )
+
+There is no direct replacement for the `Unregister<TMessage>(object, Action<TMessage>)` method in the MVVM Toolkit. 
+
+Alternatively, we recommend achieving a similar functionality with the MVVM Toolkit's `IMessenger` extension method `Unregister<TMessage>(object)`. 
```

Comment:
We might want to mention that this because a given recipient can only have a single registered handler for any given message type, so there's no need to specify what handler to remove for a message/recipient pair - there can only be up to one at any given time 😊

**Bonus point:** you can see that the MVVMLight solution here actually **creates another closure**, as it's using the method group syntax to pass the handler to remove. It's then using reflection to actually grab the underlying `MethodInfo` instance to determine what handler to remove. Our solution is much more efficient in this case, as we skip these two steps entirely (so again no reflection and 0 allocations).

Reply:
I've popped your notes in on this one. I've also updated for the additional `Unregister` method with token too.